test/e2e/daemon_set.go

```diff
@@ -427,12 +427,27 @@ func checkRunningOnAllNodes(f *framework.Framework, selector map[string]string)
 		framework.ExpectNoError(err)
 		nodeNames := make([]string, 0)
 		for _, node := range nodeList.Items {
+			if taintedNoSchedule(node) {
```

Comment:
Is there a reason to write a new function vs using ``isNodeTainted``?

https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/util.go#L2436

And are taints the only thing to consider?  What about ready or non-schedulable?  Would it make more sense to use ``GetReadySchedulableNodesOrDie`` to get a list of nodes to check?

https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/util.go#L2469